lib/platform/bitbucket-server/utils.js

```diff
@@ -17,7 +17,7 @@ const prInfo = pr => ({
   body: pr.description,
   branchName: pr.fromRef.displayId,
   title: pr.title,
-  state: prStates.closed.includes(pr.state) ? 'closed' : pr.state.toLowerCase(),
+  state: pr.state,
```

Comment:
The "state" field needs to be normalized to match with other platforms. Did you find the list of all possible states that BB server can return?

Reply:
I just grep'd the source code to confirm that we only care about open/closed/merged. I think "declined" should be counted as "closed", do you agree?

Reply:
Without this it will not find declined pr because

https://github.com/renovatebot/renovate/blob/21c0793c1ba8916396c089b1004b424ba8d377ff/lib/platform/bitbucket-server/index.js#L460-L462

https://github.com/renovatebot/renovate/blob/4fded73b62bacc0dc2b95ea953d7b8a575e81caf/lib/platform/bitbucket-server/utils.js#L8

https://github.com/renovatebot/renovate/blob/21c0793c1ba8916396c089b1004b424ba8d377ff/lib/platform/bitbucket-server/index.js#L467

causes this to fail: (`closed` is not in that array)

https://github.com/renovatebot/renovate/blob/21c0793c1ba8916396c089b1004b424ba8d377ff/lib/platform/bitbucket-server/index.js#L444-L447

Reply:
Possible states from here [/rest/api/1.0/projects/{projectKey}/repos/{repositorySlug}/pull-requests](https://docs.atlassian.com/bitbucket-server/rest/6.0.0/bitbucket-rest.html?utm_source=%2Fstatic%2Frest%2Fbitbucket-server%2Flatest%2Fbitbucket-rest.html&utm_medium=301#idp250)

are either `OPEN`, `DECLINED` or `MERGED`.

Reply:
Should i refactor the state handling?

Reply:
@ViceIce yes, please refactor if you think that's preferred. This current state handling was copied over from bitbucket cloud work and I'm definitely not tied to it

Reply:
@rarkins I've added a commit. 😃